api/src/main/java/org/openmrs/module/fhir2/api/dao/FhirAllergyIntoleranceDao.java

```diff
@@ -9,8 +9,29 @@
  */
 package org.openmrs.module.fhir2.api.dao;
 
+import java.util.List;
+
 import org.openmrs.Allergy;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+import org.openmrs.util.PrivilegeConstants;
 
 public interface FhirAllergyIntoleranceDao extends FhirDao<Allergy> {
 	
+	@Override
+	@Authorized(PrivilegeConstants.GET_ALLERGIES)
+	Allergy get(String uuid);
+	
+	@Override
+	@Authorized({ PrivilegeConstants.ADD_ALLERGIES, PrivilegeConstants.EDIT_ALLERGIES })
+	Allergy createOrUpdate(Allergy newEntry);
+	
+	@Override
+	@Authorized({ PrivilegeConstants.EDIT_ALLERGIES })
+	Allergy delete(String uuid);
```

Comment:
Ugh... so this gets complicated. I modeled the behaviour based on the privileges for Obs. So, for instance, voiding an Obs (which is all we ever do) only requires the `EDIT_OBS` privilege. The `DELETE_OBS` privilege protects the purge operation (which actually deletes the record from the database). Of course, Obs is special (since an Obs is never edited, simply voided and replaced) so this may be a special case and I need to re-evaluate this. After all, `DELETE_ALLERGIES` does seem to be used to protect the the void rather than purge operation.